webapp/src/components/common/modal/modal.tsx

```diff
@@ -0,0 +1,263 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
```

Comment:
Note to reviewer - this is a copy of `GenericModal` from MM webapp, converted to a functional component and removed some unneeded props.